algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/molecular/ProteinIsExpressedByIHC.kt

```diff
@@ -7,12 +7,16 @@ import com.hartwig.actin.datamodel.PatientRecord
 import com.hartwig.actin.datamodel.algo.Evaluation
 import com.hartwig.actin.datamodel.algo.EvaluationResult
 import com.hartwig.actin.datamodel.clinical.PriorIHCTest
+import java.time.LocalDate
 
-class ProteinIsExpressedByIHC internal constructor(private val protein: String) : EvaluationFunction {
+class ProteinIsExpressedByIHC(private val protein: String,  private val gene: String, private val maxTestAge: LocalDate? = null) : EvaluationFunction {
 
     override fun evaluate(record: PatientRecord): Evaluation {
         val ihcTests = IhcTestFilter.allIHCTestsForProtein(record.priorIHCTests, protein)
 
+        val geneIsWildType = MolecularRuleEvaluator.geneIsWildTypeForPatient(gene, record, maxTestAge)
+        val additionalMessage = if (geneIsWildType) " though $gene is wild-type in recent molecular test" else ""
```

Comment:
Minor but you could consider moving these steps into the `else` since they are only used there.